src/Pact/Native/Trans.hs

```diff
@@ -47,6 +62,12 @@ chooseFunction2 fp_double fp_musl fp_dec =
   ifExecutionFlagSet' FlagDisableNewTrans fp_double =<<
     ifExecutionFlagSet' FlagDisableNewTransDec fp_musl fp_dec
 
+getTransGasParams :: Eval e Dec.TransGasParams
+getTransGasParams = do
+  gUsed <- view eeGas >>= liftIO . readIORef
+  gLim <- view (eeGasEnv . geGasLimit)
+  pure (Dec.TransGasParams (fromIntegral gUsed) (fromIntegral gLim) chargePactArith)
```

Comment:
I see only one tiny issue here, and it's that we probably need to ensure that the free gas model doesn't make this explode and make all transcendentals fail all the time.